homeassistant/components/vallox/fan.py

```diff
@@ -171,37 +220,37 @@ async def async_turn_on(
         """Turn the device on."""
         _LOGGER.debug("Turn on: %s", speed)
 
-        # Only the case speed == None equals the GUI toggle switch being activated.
-        if speed is not None:
-            return
+        update_needed = False
 
-        if self._is_on:
-            _LOGGER.error("Already on")
-            return
+        if preset_mode:
+            update_needed = await self._async_set_preset_mode_internal(
+                preset_mode, update_state_proxy=False
+            )
 
-        try:
-            await self._client.set_values({METRIC_KEY_MODE: MODE_ON})
+        if not self._is_on:
+            try:
+                await self._client.set_values({METRIC_KEY_MODE: MODE_ON})
 
-        except OSError as err:
-            self._available = False
-            _LOGGER.error("Error turning on: %s", err)
-            return
+            except OSError as err:
+                _LOGGER.error("Error turning on: %s", err)
 
-        # This state change affects other entities like sensors. Force an immediate update that can
-        # be observed by all parties involved.
-        await self._state_proxy.async_update()
+            else:
+                update_needed = True
+
+        if update_needed:
+            # This state change affects other entities like sensors. Force an immediate update that
+            # can be observed by all parties involved.
+            await self._state_proxy.async_update()
 
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn the device off."""
         if not self._is_on:
-            _LOGGER.error("Already off")
             return
 
         try:
             await self._client.set_values({METRIC_KEY_MODE: MODE_OFF})
 
         except OSError as err:
-            self._available = False
```

Comment:
Is the idea that availability is set only in `async_update` instead?

Reply:
Yes, exactly. I realized it never made sense to have this here in the first place.